File: src/p0003/java/solution.java
class Solution {
    public int lengthOfLongestSubstring(String s) {
        int[] count = new int[256];
        int i = 0, j = 0, result = 0;
        while (j < s.length()) {
            count[s.charAt(j)] += 1;
            while (count[s.charAt(j)] > 1) {
                count[s.charAt(i)] -= 1;
                i += 1;
            }
            result = Math.max(result, j + 1 - i);
            j += 1;
        }
        return result;
    }
}

File: src/p0003/python/solution.py
class Solution:
    def lengthOfLongestSubstring(self, s: str) -> int:
        count = [0] * 256
        result, i, j = 0, 0, 0
        while j < len(s):
            count[ord(s[j])] += 1
            while count[ord(s[j])] > 1:
                count[ord(s[i])] -= 1
                i = i + 1
            result = max(result, j + 1 - i)
            j += 1
        return result
